### DISCUSSION TITLE
Replacement of deprected methods - Docs and samples missing

### DISCUSSION THREAD

**Discussion by jschmi102** [joomla/joomla-cms#38243] [Q&A]
I am in the process to cleanup the code of my extensions to fit the requirements for the joomla4 API.  
Unfortunately there are no migration documentations available to support this effort. I also could not find any references or 
implementations with the current joomla4 code. If you look into this code you will find mostly the use of deprecated (old) methods. Also searching the web did not help either. Would be great if someone could help with some hints - especially with these three topics below:      
 
**1. Create a controller instance:**
`$controller = BaseController::getInstance('my-extension');`
**The Hint for the deprecated "getInstance" is:** 
_Joomla\CMS\MVC\Controller\BaseController::getInstance(string $prefix, array $config=[]) : static
Deprecated. 5.0 Get the controller through the MVCFactory instead_
There are no docs nor any samples how to create my controller via MVCFactory

**2. add an includepath**
`HTMLHelper::addIncludePath(JPATH_COMPONENT . '/helpers/html');`
**hint for for the deprecated "addIncludePath" is:**
_Joomla\CMS\HTML\HTMLHelper::addIncludePath(string $path='') : array
Deprecated. 5.0 Use the service registry instead_
There are no docs nor any samples how to use the service registry to define my htmlhelper include path

**Get an table instance:** 
`Table::getInstance($type, $prefix, $config);`
**hint for for the deprecated "getInstance" is:**
_Joomla\CMS\Table\Table::getInstance(string $type, string $prefix='JTable', array $config=[]) : Table|boolean
Deprecated. 5.0 Use the MvcFactory instead_
There are no docs nor any samples how to use MvcFactory instead of getInstance 

Many thanks in advance!

**Comment by stephane2049** [comment]
Hello,
Did you find any solution for your questions, especially for topic #2 ? 
I'm in the very same process of reviewing all my code for adapting it to joomla4 and i'm still facing the same difficulties since i can't find more documentation 4 months after your message.

**Reply by joomdonation** [reply]
@stephane2049 For the topic #2, we can learn from code of core components like com_contact

- Add your service https://github.com/joomla/joomla-cms/blob/4.2-dev/administrator/components/com_contact/src/Service/HTML/AdministratorService.php
- Register that service in extension boot method https://github.com/joomla/joomla-cms/blob/4.2-dev/administrator/components/com_contact/src/Extension/ContactComponent.php#L71
- Call that service https://github.com/joomla/joomla-cms/blob/4.2-dev/administrator/components/com_contact/tmpl/contacts/default.php#L130

If you need more detailed explanation, I think @laoneo could explain more details about this.
